guide/kong_integration.md

```diff
@@ -0,0 +1,196 @@
+# Using Kong as an API gateway for OpenFaaS
+
+[Kong](https://getkong.org) is an API gateway that provides features such as security, logging, and rate limiting. By putting this in front of OpenFaaS you can quickly get access to these things and a lot more via [the many other plugins written](https://getkong.org/plugins/) for it.
+
+Below is a demo of how you could use Kong as an authentication layer for OpenFaaS.
+
+## Setup OpenFaaS
+
+```
+$ docker swarm init --advertise-addr $(hostname -i)
+
+$ git clone https://github.com/alexellis/faas && \
+  cd faas && \
+  ./deploy_stack.sh
+```
+
+Check that one of the sample functions works
+
+```
+$ curl localhost:8080/function/func_echoit -d 'hello world'
+hello world
+```
+
+## Setup Kong
+```
+$ docker service create --network func_functions --detach=false \
+      --name kong-database \
+      -p 5432:5432 \
```

Comment:
I don't think you need to expose this, both the `kong` and `kong-migrations` containers can access this without the mapping over the `func_functions` network.

Reply:
The network I believe is allowing us to specify the upstream by DNS name inside the overlay network such as `http://gateway:8080/`. Or what were you thinking?

Reply:
was that question for me?

Reply:
Yes - didn't understand your comment re: network mapping

Reply:
ah ok, yes the migration and main kong services are accessing the database over the `func_functions` network via the dns name. There is no need to expose the port as its only being used internally.

Reply:
sure. I will not expose 5432. its not needed.